rpc/api.go

```diff
@@ -4,269 +4,454 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
-	"math/big"
-	"strconv"
-	"strings"
 
 	"github.com/dontpanicdao/caigo"
 	"github.com/dontpanicdao/caigo/types"
 )
 
-type Events struct {
-	Events []Event `json:"events"`
-}
-
-type Event struct {
-	*types.Event
-	FromAddress     string `json:"from_address"`
-	BlockHash       string `json:"block_hash"`
-	BlockNumber     int    `json:"block_number"`
-	TransactionHash string `json:"transaction_hash"`
-}
-
-type EventParams struct {
-	FromBlock  uint64 `json:"fromBlock"`
-	ToBlock    uint64 `json:"toBlock"`
-	PageSize   uint64 `json:"page_size"`
-	PageNumber uint64 `json:"page_number"`
-}
-
 // Call a starknet function without creating a StarkNet transaction.
-func (sc *Client) Call(ctx context.Context, call types.FunctionCall, hash string) ([]string, error) {
+func (sc *Client) Call(ctx context.Context, call types.FunctionCall, blockIDOption BlockIDOption) ([]string, error) {
+	opt := &blockID{}
+	err := blockIDOption(opt)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
```suggestion
	if err := blockIDOption(opt); err != nil {
		return nil, err
	}
```

nit: minimize err scope

Reply:
applied